src/replicache.ts

```diff
@@ -613,6 +531,114 @@ export default class Replicache implements ReadTransaction {
     }
   }
 
+  private async _wrapInOnlineCheck(
+    f: () => Promise<void>,
+    name: string,
+  ): Promise<void> {
+    let online = true;
+
+    try {
+      await f();
+    } catch (e) {
+      // The error paths of beginPull and maybeEndPull need to be reworked.
+      //
+      // We want to distinguish between:
+      // a) network requests failed -- we're offline basically
+      // b) sync was aborted because one's already in progress
+      // c) oh noes - something unexpected happened
+      //
+      // Right now, all of these come out as errors. We distinguish (b) with a
+      // hacky string search. (a) and (c) are not distinguishable currently
+      // because repc doesn't provide sufficient information, so we treat all
+      // errors that aren't (b) as (a).
+      if (e.toString().includes('JSLogInfo')) {
+        online = false;
+      }
+      console.info(`${name} returned: ${e}`);
+    }
+
+    this._online = online;
+  }
+
+  /**
+   * Push pushes pending changes to the [[pushURL]].
+   */
+  async push(): Promise<void> {
+    await this._wrapInOnlineCheck(() => this._push(MAX_REAUTH_TRIES), 'Push');
+  }
+
+  private async _push(maxAuthTries: number): Promise<void> {
+    const pushResponse = await this._invoke('tryPush', {
+      pushURL: this._pushURL,
+      pushAuth: this._pushAuth,
+    });
+
+    let reauth = false;
+
+    const {httpRequestInfo} = pushResponse;
+
+    if (httpRequestInfo) {
+      reauth = checkStatus(httpRequestInfo, 'push', this._pushURL);
+      // mutationInfos support was never added to repc. We used to log all the
```

Comment:
I don't understand this comment. Remove?

Reply:
Or change it to a todo. It is not clear if we want the old replicache-client behavior